Composer/packages/client/src/components/ProjectTree/TriggerCreationModal.tsx

```diff
@@ -299,29 +316,50 @@ export const TriggerCreationModal: React.FC<TriggerCreationModalProps> = (props)
     if (option) {
       const errors: TriggerFormDataErrors = {};
       errors.event = validateEventKind(selectedType, option.key as string);
-      setFormData({ ...formData, $kind: option.key as string, errors: { ...formData.errors, ...errors } });
+      setFormData({
+        ...formData,
+        $kind: option.key as string,
+        errors: { ...formData.errors, ...errors },
+      });
     }
   };
 
   const onNameChange = (e, name) => {
     const errors: TriggerFormDataErrors = {};
     errors.intent = validateIntentName(selectedType, name);
-    if (showTriggerPhrase) {
+    if (showTriggerPhrase && formData.triggerPhrases) {
       errors.triggerPhrases = getLuDiagnostics(name, formData.triggerPhrases);
     }
-    setFormData({ ...formData, intent: name, errors: { ...formData.errors, ...errors } });
+    setFormData({
+      ...formData,
+      intent: name,
+      errors: { ...formData.errors, ...errors },
+    });
   };
 
   const onChangeRegEx = (e, pattern) => {
     const errors: TriggerFormDataErrors = {};
     errors.regEx = validateRegExPattern(selectedType, isRegEx, pattern);
-    setFormData({ ...formData, regEx: pattern, errors: { ...formData.errors, ...errors } });
+    setFormData({
+      ...formData,
+      regEx: pattern,
+      errors: { ...formData.errors, ...errors },
+    });
   };
 
+  //Trigger phrase is optional
   const onTriggerPhrasesChange = (body: string) => {
     const errors: TriggerFormDataErrors = {};
-    errors.triggerPhrases = getLuDiagnostics(formData.intent, body);
-    setFormData({ ...formData, triggerPhrases: body, errors: { ...formData.errors, ...errors } });
+    if (body) {
+      errors.triggerPhrases = getLuDiagnostics(formData.intent, body);
+    } else {
+      errors.triggerPhrases = '';
+    }
```

Comment:
There are three appearances of checking the triggerPhrases existence.
Optimize it by returning undefined in getLuDiagnostics() when triggerPhrases is null.